### PR TITLE
Fix out of sources tree

### PR DESCRIPTION
**Pull request by remicollet**
Notice: "pecl install awscrt" build the extension out of the sources tree

### CODE REVIEW

File: Makefile.frag [hunk 1]
```diff
@@ -58,17 +58,18 @@ extension: $(builddir)/ext/awscrt.lo
 # Force the crt object target to depend on the CRT static library
 $(builddir)/ext/awscrt.lo: $(builddir)/ext/awscrt.c
 
-$(builddir)/ext/awscrt.c: $(CMAKE_BUILD_DIR)/aws-crt-ffi-static/libaws-crt-ffi.a $(builddir)/ext/api.h $(builddir)/ext/awscrt_arginfo.h
+$(builddir)/ext/awscrt.c: $(CMAKE_BUILD_DIR)/aws-crt-ffi-static/libaws-crt-ffi.a $(srcdir)/ext/api.h $(srcdir)/ext/awscrt_arginfo.h
```
Comment (remicollet) [inline Makefile.frag:61]: These files are generated in the sources tree, so fix paths

File: Makefile.frag [hunk 2]
```diff
@@ -58,17 +58,18 @@ extension: $(builddir)/ext/awscrt.lo
...
 
-$(builddir)/ext/awscrt_arginfo.h: $(srcdir)/ext/awscrt.stub.php $(srcdir)/gen_stub.php
+$(srcdir)/ext/awscrt_arginfo.h: $(srcdir)/ext/awscrt.stub.php $(srcdir)/gen_stub.php
 ifeq ($(GENERATE_STUBS),1)
 	# generate awscrt_arginfo.h
-	mkdir -p $(builddir)/ext && php $(srcdir)/gen_stub.php --minimal-arginfo $(srcdir)/ext/awscrt.stub.php
+	$(PHP_EXECUTABLE) $(srcdir)/gen_stub.php --minimal-arginfo $(srcdir)/ext/awscrt.stub.php
```
Comment (remicollet) [inline Makefile.frag:66]: `php` command may be different than the target build or missing

File: config.m4 [hunk 3]
```diff
@@ -26,5 +26,6 @@ if test "$PHP_AWSCRT" != "no"; then
     # Sources for the PHP extension itself
     AWSCRT_SOURCES=ext/awscrt.c
     PHP_NEW_EXTENSION(awscrt, $AWSCRT_SOURCES, $ext_shared)
+    PHP_ADD_BUILD_DIR($ext_builddir/ext)
```
Comment (remicollet) [inline config.m4:29]: Create the build directory to avoid `mkdir` above

**Comment by TingDaoK** [comment]
Can you please explain a bit more about what issue this is trying to resolve?

**Comment by remicollet** [comment]
> Can you please explain a bit more about what issue this is trying to resolve?

# Context

I'm building this extension as RPM for users of my [repository](https://rpms.remirepo.net/) 
which is quite commonly used by Fedora, CentOS, RHEL, and clones

See the [extension list](https://blog.remirepo.net/pages/PECL-extensions-RPM-status)

My repo is used as the upstream for official Fedora packages (I also maintain most of them) and so for RHEL packages.

# Problem

I noticed that the arginfo header is re-generated on each build, which implies downloading nikic/php-parser, a blocker for RPM builds, done offline, without internet access.

So I dig in the config.m4 and found some issues especially related to out-of-sources-tree build (`$srcdir != $ext_builddir`), see inline comments.

Direct `php` call is also a blocker when the `php` command is not available in the path, or when a different version than the target of the build.

See: https://github.com/php/php-src/blob/php-8.3.13/build/Makefile.global#L153-L159
`$(PHP)` must be used for php-src itself and `$(PHP_EXECUTABLE)` for extension

**Review by TingDaoK — APPROVED** [review]

**Pull request merged**